src/proxy/http/classify.rs

```diff
@@ -24,28 +30,30 @@ pub trait ClassifyResponse {
     /// A response classification.
     type Class;
     type Error;
+    type ClassifyEos: ClassifyEos<Class = Self::Class, Error = Self::Error>;
 
-    /// Update the classifier with the response headers.
+    /// Produce a stream classifier for this response.
     ///
     /// If this is enough data to classify a response, a classification may be
-    /// returned. Implementations should expect that `end` or `error` may be
-    /// called even when a class is returned.
-    ///
-    /// This is expected to be called only once.
-    fn start<B>(&mut self, headers: &http::Response<B>) -> Option<Self::Class>;
+    /// returned.
+    fn start<B>(self, headers: &http::Response<B>) -> (Self::ClassifyEos, Option<Self::Class>);
```

Comment:
TIOLI: Maybe instead of a tuple, this should return a struct that more clearly encodes the states that the return type represents? Something like
```rust
pub struct ResponseStart<C: ClassifyResponse> {
    classify_eos: C::ClassifyEos,
    class_at_start: Option<C::Class>
}
```

On the other hand, maybe that just complicates things... 🤷‍♀️

Reply:
Yeah, I think it's OK as-is, for now. We can see how we like it and change it later if desirable.